### PR TITLE
Modifications to support geotileplus generation in setsm-utils

### PR DESCRIPTION
**Pull request by clairecporter**
Plus python3 upgrades

### CODE REVIEW

**Comment by clairecporter** [comment]
@ehusby , I had to make a few modifications in dem.py to support density calculation outside the class.  All my test still pass though, so I am fairly sure the changes are invisible for any code that uses those classes.

File: lib/dem.py [hunk 1]
```diff
@@ -127,15 +133,11 @@ def __init__(self,metapath,md=None):
             self.metapath = metapath
             self.lsf_dem = os.path.join(self.srcdir,self.sceneid+"_dem_smooth.tif")
             self.dem = os.path.join(self.srcdir,self.sceneid+"_dem.tif")
-            self.dem_edge_masked = os.path.join(self.srcdir,self.sceneid+"_dem_edge-masked.tif")
```
Comment (ehusby) [inline lib/dem.py:130]: Looks like you might've accidentally removed this and the additional small block below that I added in https://github.com/PolarGeospatialCenter/pgcdemtools/pull/19?

File: lib/utils.py [hunk 2]
```diff
@@ -384,46 +378,3 @@ def drange(start, stop, step):
     while r < stop:
         yield r
         r += step
-
```
Comment (ehusby) [inline lib/utils.py:387]: Same thing as the other, was this code from https://github.com/PolarGeospatialCenter/pgcdemtools/commit/d3884c8ecfa7a61d9081ea28b4ec425b56dc1124 accidentally removed?

**Review by ehusby — CHANGES_REQUESTED** [review]

**Comment by clairecporter** [comment]
Yup. My mistake. Will fix it.

**Comment by clairecporter** [comment]
This latest commit readded those lines and also the changes in utils.py that I also reverted.  Apologies again.

**Comment by clairecporter** [comment]
Wait there are some more changes that I missed.  give me a sec.

File: lib/dem.py [hunk 3]
```diff
@@ -1818,3 +1704,30 @@ def format_as_imd(contents):
...
+def get_matchtag_density(matchtag, geom_area=None):
+    ds = gdal.Open(matchtag)
+    b = ds.GetRasterBand(1)
+    gtf = ds.GetGeoTransform()
+    matchtag_res_x = gtf[1]
+    matchtag_res_y = gtf[5]
+    matchtag_size_x = ds.RasterXSize
+    matchtag_size_y = ds.RasterYSize
+    matchtag_ndv = b.GetNoDataValue()
+    data = utils.gdalReadAsArraySetsmSceneBand(b)
```
Comment (ehusby) [inline lib/dem.py:1718]: Got it! I'd messed up my test merge and got confused when looking at the working tree diff

**Review by ehusby — APPROVED** [review]
A+ on acing the non-proper way to resolve a merge goof, which in my opinion is way harder! ;)

**Comment by clairecporter** [comment]
Thanks for the good grade!

**Pull request merged**